main.tf

```diff
@@ -83,10 +86,19 @@ resource "aws_iam_role_policy" "user" {
 resource "aws_transfer_user" "default" {
   for_each = var.users
 
-  user_name      = each.key
-  home_directory = each.value.home_directory
-  role           = aws_iam_role.user[each.key].arn
-  server_id      = aws_transfer_server.default.id
+  user_name           = each.key
+  home_directory      = var.restricted_mode ? null : each.value.home_directory
+  home_directory_type = var.restricted_mode ? "LOGICAL" : "PATH"
+  role                = aws_iam_role.user[each.key].arn
+  server_id           = aws_transfer_server.default.id
+  tags                = var.tags
+  dynamic "home_directory_mappings" {
```

Comment:
please add a blank line between tags and dynamic: https://www.terraform.io/language/syntax/style